tests/unit/test_profiling_integration.py

```diff
@@ -0,0 +1,107 @@
+import json
+from unittest.mock import patch
+
+import pytest
+from ops.testing import Container, Relation, State
+
+from charms.tls_certificates_interface.v4.tls_certificates import (
+    TLSCertificatesRequiresV4,
+    Certificate,
+)
+from tests.unit.helpers import get_otelcol_file
+
+
+@pytest.fixture
+def tls_mock(cert_obj, private_key):
+    with patch.object(
+        TLSCertificatesRequiresV4, "_find_available_certificates", return_value=None
+    ), patch.object(
+        TLSCertificatesRequiresV4, "get_assigned_certificate", return_value=(cert_obj, private_key)
+    ), patch.object(Certificate, "from_string", return_value=cert_obj):
+        yield
+
+
+@pytest.mark.parametrize("insecure_skip_verify", (True, False))
+def test_send_profiles_integration(ctx, insecure_skip_verify, unit_name, config_folder):
+    """Scenario: a profiling relation joined and sent us a grpc endpoint."""
+    # GIVEN otelcol deployed in isolation
+
+    pyro_url = "my.fqdn.cluster.local:12345"
+    # WHEN a profiling relation joins and pyroscope sent an endpoint
+    send_profiles = Relation(
+        endpoint="send-profiles",
+        remote_app_data={
+            "otlp_grpc_endpoint_url": json.dumps(pyro_url),
+            "otlp_http_endpoint_url": json.dumps("foobar"),
+        }
+    )
+    state_in = State(relations=[send_profiles],
+                     config={"tls_insecure_skip_verify": insecure_skip_verify})
+    ctx.run(ctx.on.update_status(), state=state_in)
+
+    # THEN the profiling pipeline contains an exporter to the expected url
```

Comment:
We also need an assertion on the config file from before the relation joined.

Reply:
uhm asserting what precisely? can you clarify?